lib/node_modules/@stdlib/math/base/special/gcd/src/main.c

```diff
@@ -0,0 +1,181 @@
+		return a;
+	}
+	// Reduce `a` and/or `b` to odd numbers and keep track of the greatest power of 2 dividing both `a` and `b`...
+	while ( ac%2 == 0 && bc%2 == 0 ) {
+		ac /= 2; // right shift
+		bc /= 2; // right shift
+		k *= 2; // left shift
+	}
+	// Reduce `a` to an odd number...
+	while ( ac%2 == 0 ) {
+		ac /= 2; // right shift
+	}
+	// Henceforth, `a` is always odd...
+	while ( bc ) {
+		// Remove all factors of 2 in `b`, as they are not common...
+		while ( bc%2 == 0 ) {
+			bc /= 2; // right shift
+		}
+		// `a` and `b` are both odd. Swap values such that `b` is the larger of the two values, and then set `b` to the difference (which is even)...
+		if ( ac > bc ) {
+			t = bc;
+			bc = ac;
+			ac = t;
+		}
+		bc -= ac; // b=0 iff b=a
+	}
+	// Restore common factors of 2...
+	return k * ac;
+}
+
+/**
+* Computes the greatest common divisor (gcd) using the binary GCD algorithm and bitwise operations.
+*
+* @param a    input value
+* @param b    input value
+* @return	  output value
+*
+* @example
+* double out = bitwise( 48.0, 18.0 );
+* // returns 6.0
+*/
+
+double bitwise( const double a, const double b ) {
+	int64_t k = 0;
+	int64_t t;
+	int64_t ac;
+	int64_t bc;
+
+	ac = (int64_t)a;
+	bc = (int64_t)b;
+
+	// Simple cases:
+	if ( a == 0.0 ) {
+		return b;
+	}
+	if ( b == 0.0 ) {
+		return a;
+	}
+	// Reduce `a` and/or `b` to odd numbers and keep track of the greatest power of 2 dividing both `a` and `b`...
+	while ( (ac & 1) == 0 && (bc & 1) == 0 ) {
+		ac >>= 1; // right shift
+		bc >>= 1; // right shift
+		k += 1;
+	}
+	// Reduce `a` to an odd number...
+	while ( (ac & 1) == 0 ) {
+		ac >>= 1; // right shift
+	}
+	// Henceforth, `a` is always odd...
+	while ( bc ) {
+		// Remove all factors of 2 in `b`, as they are not common...
+		while ( (bc & 1) == 0 ) {
+			bc >>= 1; // right shift
+		}
+		// `a` and `b` are both odd. Swap values such that `b` is the larger of the two values, and then set `b` to the difference (which is even)...
+		if ( ac > bc ) {
+			t = bc;
+			bc = ac;
+			ac = t;
+		}
+		bc -= ac; // b=0 iff b=a
+	}
+	// Restore common factors of 2...
+	return ac << k;
+}
+
+/**
+* Computes the greatest common divisor (gcd).
+*
+* @param a    input value
+* @param b    input value
+* @return	  output value
+*
+* @example
+* double out = stdlib_base_gcd( 48.0, 18.0 );
+* // returns 6.0
+*/
+
+double stdlib_base_gcd( const double a, const double b ) {
+	double ac;
+	double bc;
+
+	ac = a;
+	bc = b;
+	if ( stdlib_base_is_nan( a ) || stdlib_base_is_nan( b ) ) {
+		return 0.0/0.0;
+	}
+	if (
+		a == STDLIB_CONSTANT_FLOAT64_PINF ||
+		b == STDLIB_CONSTANT_FLOAT64_PINF ||
+		a == STDLIB_CONSTANT_FLOAT64_NINF ||
+		b == STDLIB_CONSTANT_FLOAT64_NINF
+	) {
+		return 0.0/0.0;
+	}
+	if ( !( stdlib_base_is_integer( a ) && stdlib_base_is_integer( b ) ) ) {
+		return 0.0/0.0;
+	}
+	if ( ac < 0 ) {
+		ac = -ac;
+	}
+	if ( bc < 0 ) {
+		bc = -bc;
+	}
+	if ( ac <= STDLIB_CONSTANT_INT32_MAX && bc <= STDLIB_CONSTANT_INT32_MAX ) {
```

Comment:
We don't have a separate package for this, but, in this case, you should define a `static const` with the maximum 64-bit integer value and then update this condition accordingly.